drivers/spi/Kconfig.nrfx

```diff
@@ -6,6 +6,7 @@ menuconfig SPI_NRFX
 	default y
 	depends on SOC_FAMILY_NORDIC_NRF
 	depends on MULTITHREADING
+	depends on GPIO
```

Comment:
> Now we will get:
> 
> * A CMake warning will be issued that the SPI driver is not compiled in.
> * A linker error in `spi_nor.c` failing to reference a SPI driver.
> 
> With this it becomes more obvious that SPI driver is enabled by default because of the external flash mounted on the DK.

I'm afraid it might still not be obvious enough. The linker error is quite cryptic, as it says about undefined reference to `__device_dts_ord_*`. How do you know that the device is SPI?
Hence, I think it should be `select GPIO`. Then, for `CONFIG_GPIO=n`, we'll get a Kconfig warning that `GPIO` got the value `y`, although it was supposed to be `n`. And in `menuconfig`, one can find out what exactly makes this option set to `y`.